contracts/hyphen/LiquidityPool.sol

```diff
@@ -180,23 +197,65 @@ contract LiquidityPool is
         uint256 amount,
         string calldata tag
     ) public tokenChecks(tokenAddress) whenNotPaused nonReentrant {
+        address sender = _msgSender();
+        uint256 rewardAmount = _depositErc20(sender, toChainId, tokenAddress, receiver, amount);
+
+        // Emit (amount + reward amount) in event
+        emit Deposit(sender, tokenAddress, receiver, toChainId, amount + rewardAmount, rewardAmount, tag);
+    }
+
+    /**
+     * @dev Function used to deposit tokens into pool to initiate a cross chain token swap And transfer .
+     * @param toChainId Chain id where funds needs to be transfered
+     * @param tokenAddress ERC20 Token address that needs to be transfered
+     * @param receiver Address on toChainId where tokens needs to be transfered
+     * @param amount Amount of token being transfered
+     * @param tag Dapp unique identifier
+     * @param swapRequest information related to token swap on exit chain
+     */
+    function depositAndSwapErc20(
+        address tokenAddress,
+        address receiver,
+        uint256 toChainId,
+        uint256 amount,
+        string calldata tag,
+        SwapRequest[] calldata swapRequest
+    ) public tokenChecks(tokenAddress) whenNotPaused nonReentrant {
+    
+        uint256 totalPercentage = 0;
+        for (uint256 index = 0; index < swapRequest.length; ++index) {
```

Comment:
create a separate variable for swapRequest.length, and enclose the for loop inside unchecked {} to save gas